File: java/org.financer.client/src/main/java/org/financer/client/domain/model/user/Setting.java
package org.financer.client.domain.model.user;

import org.financer.shared.domain.model.value.objects.SettingPair;

public class Setting {

    private long id;
    private User user;
    private SettingPair pair;

    public Setting setValue(String value) {
        this.setPair(this.getPair().setValue(value));
        return this;
    }

    /*
     * Getters and Setters
     */

    public long getId() {
        return id;
    }

    public Setting setId(long id) {
        this.id = id;
        return this;
    }

    public User getUser() {
        return user;
    }

    public Setting setUser(User user) {
        this.user = user;
        return this;
    }

    public SettingPair getPair() {
        return pair;
    }

    public Setting setPair(SettingPair pair) {
        this.pair = pair;
        return this;
    }
}

File: java/org.financer.client/src/main/kotlin/org/financer/client/domain/model/user/Setting.kt
package org.financer.client.domain.model.user

import org.financer.shared.domain.model.value.objects.SettingPair

class Setting {
    /*
     * Getters and Setters
     */ var id: Long = 0
        private set
    var user: User? = null
        private set
    var pair: SettingPair? = null
        private set

    fun setValue(value: String?): Setting {
        setPair(pair!!.setValue(value))
        return this
    }

    fun setId(id: Long): Setting {
        this.id = id
        return this
    }

    fun setUser(user: User?): Setting {
        this.user = user
        return this
    }

    fun setPair(pair: SettingPair?): Setting {
        this.pair = pair
        return this
    }
}